src/main/java/main/java/duke/Ui.java

```diff
@@ -0,0 +1,138 @@
+package main.java.duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Ui {
+
+    private TaskList taskList = null;
+    private Storage storage = null;
+    private Parser parser = new Parser();
+
+    /**
+     * Creates an instance of the Ui class
+     * @param taskList taskList object used to store tasks
+     * @param storage storage object used to read and write for the tasklist.txt file
+     */
+    public Ui(TaskList taskList, Storage storage) {
+        this.taskList = taskList;
+        this.storage = storage;
+    }
+
+    /**
+     * Prints the DukeException error message.
+     * @param errMsg
+     */
+    public String invalidInput(String errMsg) {
+        DukeException exception = new DukeException(errMsg);
+        return exception.getMessage();
+    }
+
+    /**
+     * Prints a horizontal line of fixed length.
+     */
+    public void horiLine(int length) {
+        for (int i = 0; i < length; i++) {
+            System.out.print("-");
+            if (i == length - 1) System.out.println("");
+        }
+    }
+
+    /**
+     * Returns a string to be displayed as the response on CLI or GUI
+     * @param nextInput user input string
+     * @return a string to be displayed as the response on CLI or GUI
+     */
+    public String getResultFromParser(String nextInput) {
+        String[] commandComponents = nextInput.split(" ", 2);
```

Comment:
Perhaps all of this can be handled by the parser as well, as these are all commands?

Reply:
Yes perhaps, the commandComponents is used to recognise the type of command. Thank you!